From 33e066c2d61931f4897410ed9adcb605c28decfc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?D=2E=20Andrei=20M=C4=83ce=C8=99?=
 <andrei@unlegacy-android.org>
Date: Sat, 1 Sep 2018 21:33:33 +0300
Subject: [PATCH 9/9] msm8960/74/94: Move GRALLOC_USAGE_PRIVATE_UNCACHED
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change I7f4174581e24e361577640b9263514a168ed482d in frameworks/native
introduces validation for Gralloc usage bits, which blocks arbitrary
private usage flags, such as GRALLOC_USAGE_PRIVATE_UNCACHED (1 << 25).

E Gralloc2: buffer descriptor contains invalid usage bits 0x2000000

While Gralloc1 has many more GRALLOC1_CONSUMER_USAGE_PRIVATE_[4-19], we
are pretty limited with our Gralloc0 headers, and 32-bit usage flags.
However, 1 << 2, 1 << 3, 1 << 6, and 1 << 7 seem to be available,
and fall under GRALLOC_USAGE_SW_{READ,WRITE}_MASK, so use them.

Change-Id: I39ffb14e4b67191f1c12501cd02350fb2eac31a6
Signed-off-by: D. Andrei Măceș <andrei@unlegacy-android.org>
---
 msm8960/libgralloc/gralloc_priv.h | 2 +-
 msm8974/libgralloc/gralloc_priv.h | 2 +-
 msm8994/libgralloc/gralloc_priv.h | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/msm8960/libgralloc/gralloc_priv.h b/msm8960/libgralloc/gralloc_priv.h
index d9f4f4ff6..5417ddb23 100644
--- a/msm8960/libgralloc/gralloc_priv.h
+++ b/msm8960/libgralloc/gralloc_priv.h
@@ -52,7 +52,7 @@ enum {
 
     /* Set this for allocating uncached memory (using O_DSYNC)
      * cannot be used with noncontiguous heaps */
-    GRALLOC_USAGE_PRIVATE_UNCACHED        =       0x02000000,
+    GRALLOC_USAGE_PRIVATE_UNCACHED        =       0x00000004,
 
     /* Buffer content should be displayed on an external display only */
     GRALLOC_USAGE_PRIVATE_EXTERNAL_ONLY   =       0x08000000,
diff --git a/msm8974/libgralloc/gralloc_priv.h b/msm8974/libgralloc/gralloc_priv.h
index 94d4334e2..67c6d199f 100644
--- a/msm8974/libgralloc/gralloc_priv.h
+++ b/msm8974/libgralloc/gralloc_priv.h
@@ -52,7 +52,7 @@ enum {
 
     /* Set this for allocating uncached memory (using O_DSYNC)
      * cannot be used with noncontiguous heaps */
-    GRALLOC_USAGE_PRIVATE_UNCACHED        =       0x02000000,
+    GRALLOC_USAGE_PRIVATE_UNCACHED        =       0x00000004,
 
     /* Buffer content should be displayed on an external display only */
     GRALLOC_USAGE_PRIVATE_EXTERNAL_ONLY   =       0x08000000,
diff --git a/msm8994/libgralloc/gralloc_priv.h b/msm8994/libgralloc/gralloc_priv.h
index 396d3e0ae..c2e84ad08 100755
--- a/msm8994/libgralloc/gralloc_priv.h
+++ b/msm8994/libgralloc/gralloc_priv.h
@@ -54,7 +54,7 @@ enum {
 
     /* Set this for allocating uncached memory (using O_DSYNC)
      * cannot be used with noncontiguous heaps */
-    GRALLOC_USAGE_PRIVATE_UNCACHED        =       0x02000000,
+    GRALLOC_USAGE_PRIVATE_UNCACHED        =       0x00000004,
 
     /* Buffer content should be displayed on an primary display only */
     GRALLOC_USAGE_PRIVATE_INTERNAL_ONLY   =       0x04000000,
-- 
2.17.1

